internal/aws/client/client.go

```diff
@@ -0,0 +1,104 @@
+package client
+
+import (
+	"context"
+	"github.com/Falco-Talon/falco-talon/configuration"
+	"github.com/Falco-Talon/falco-talon/utils"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/lambda"
+)
+
+type AWSClient struct {
+	cfg aws.Config
+
+	mu           sync.Mutex // Protects the fields below
+	lambdaClient *lambda.Client
+	//s3Client     *s3.Client
+}
+
+var (
+	awsClient *AWSClient
+	once      sync.Once
+)
+
+func Init() error {
+	var initErr error
+
+	awsConfig := configuration.GetConfiguration().AwsConfig
+
+	once.Do(func() {
+		var cfg aws.Config
+		var err error
+
+		if awsConfig.AccessKey != "" && awsConfig.SecretKey != "" && awsConfig.Region != "" {
+			cfg, err = config.LoadDefaultConfig(
+				context.TODO(),
+				config.WithRegion(awsConfig.Region),
+				config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(awsConfig.AccessKey, awsConfig.SecretKey, "")),
+			)
+		} else {
+			cfg, err = config.LoadDefaultConfig(context.TODO())
+		}
+		if err != nil {
+			initErr = err
+			return
+		}
+
+		if awsConfig.RoleArn != "" {
+			stsClient := sts.NewFromConfig(cfg)
+			assumeRoleOptions := func(o *stscreds.AssumeRoleOptions) {
+				if awsConfig.ExternalId != "" {
+					o.ExternalID = aws.String(awsConfig.ExternalId)
+				}
+			}
+			provider := stscreds.NewAssumeRoleProvider(stsClient, awsConfig.RoleArn, assumeRoleOptions)
+			cfg.Credentials = aws.NewCredentialsCache(provider)
+		}
+
+		// Perform a dry run to validate credentials
+		stsClient := sts.NewFromConfig(cfg)
+		_, err = stsClient.GetCallerIdentity(context.TODO(), &sts.GetCallerIdentityInput{})
```

Comment:
For the docs, we need to remind the IAM policy must allow `sts:get-caller-identity`

Reply:
Yes I thought the same when adding this, I wasn't able to think in a better way to properly initialize the customer in the init function in order to fail-fast.